aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -472,4 +476,111 @@ class RealAWSCognitoAuthPluginTest {
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+
+        val currentAuthState = mockk<AuthState> {
+            every { authNState } returns AuthenticationState.SignedIn(mockk())
+            every { authZState } returns AuthorizationState.SessionEstablished(credentials)
+        }
+        every { authStateMachine.getCurrentState(captureLambda()) } answers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        coEvery {
+            authService.cognitoIdentityProviderClient?.updateUserAttributes(any<UpdateUserAttributesRequest>())
+        } returns UpdateUserAttributesResponse.invoke {
+        }
+
+        every {
+            onSuccess.accept(any<AuthUpdateAttributeResult>())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.updateUserAttribute(
+            AuthUserAttribute(AuthUserAttributeKey.email(), "test@test.com"),
+            onSuccess,
+            onError
+        )
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
+        coVerify(exactly = 1) { onSuccess.accept(any<AuthUpdateAttributeResult>()) }
+        coVerify(exactly = 0) { onError.accept(any()) }
+    }
+
+    @Test
+    fun `update user attributes with success`() {
+        // GIVEN
+        val onSuccess = mockk<Consumer<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>>(relaxed = true)
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+
+        val currentAuthState = mockk<AuthState> {
+            every { authNState } returns AuthenticationState.SignedIn(mockk())
+            every { authZState } returns AuthorizationState.SessionEstablished(credentials)
+        }
+        every { authStateMachine.getCurrentState(captureLambda()) } answers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        coEvery {
+            authService.cognitoIdentityProviderClient?.updateUserAttributes(any<UpdateUserAttributesRequest>())
+        } returns UpdateUserAttributesResponse.invoke {
+        }
+
+        every {
+            onSuccess.accept(any<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.updateUserAttributes(
+            mutableListOf(
+                AuthUserAttribute(AuthUserAttributeKey.email(), "test@test.com"),
+                AuthUserAttribute(AuthUserAttributeKey.nickname(), "test")
+            ),
+            AuthUpdateUserAttributesOptions.defaults(),
+            onSuccess,
+            onError
+        )
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
+        coVerify(exactly = 1) { onSuccess.accept(any<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>()) }
```

Comment:
Can we also have a test where actual data of `Consumer<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>` is sent back rather than `any`

Reply:
not sure what you mean. The any is saying that the return type must be of MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>. Since we are not calling the cognito api, I am not sure what you meant by actual data.

Reply:
I think @sktimalsina meant that instead of mocking Cognito API response as empty `UpdateUserAttributesResponse.invoke {}` object, return some valid response and test whether Amplify API's response matches the expected response.